### ISSUE TITLE
const functions should not trigger or_fun_call 

### ISSUE THREAD

**Issue by Sciencentistguy** [rust-lang/rust-clippy#6943]
Lint name: or_fun_call

I tried this code:

```rust
let x = None;
let s: &str = "hello world";
s.unwrap_or(s.len());
```

I expected to see this happen: As str::len is a const function, it is inlined as a usize. So the suggestion to use `.unwrap_or_else()` introduces more overhead than is needed.

Instead, this happened: The lint is triggered

### Meta

- `cargo clippy -V`: `clippy 0.1.52 (f5d8117 2021-03-16)`
- `rustc -Vv`:
```
rustc 1.52.0-nightly (f5d8117c3 2021-03-16)
binary: rustc
commit-hash: f5d8117c338a788bd24abec733fd143dfceb25a0
commit-date: 2021-03-16
host: x86_64-unknown-linux-gnu
release: 1.52.0-nightly
LLVM version: 12.0.0
```

**Comment by camsteffen** [comment]
We cannot simply ignore const functions since they are not guaranteed to be evaluated at compile time. See #1527. However we can ignore `str::len` since we also ignore `slice::len`.
